REACT_STYLEGUIDE.MD

```diff
@@ -45,14 +44,31 @@ Bad:
     };
   },
   onClick() {
-    this.setState({ clickCount: this.state.clickCount++ });
+    this.setState(prevState => ({
+      clickCount: prevState.clickCount + 1
+    }));
   },
-  render({ onClick }) {
+  render() {
     return <button className="kuiButton" onClick={this.onClick}/>
   }
 });
 ```
 
+### When a state change only involves the previous state, pass setState a function instead of an object.
```

Comment:
Agreed here. I don't think it's a necessarily common occurrence, but it sounds like not understanding the queue aspect of `.setState` could lead to some extremely hard to track down bugs in the future. 

++ on this rule